### PR TITLE
Allow using -D hxcs.DEFINE and add documentation

### PR DESCRIPTION
**Pull request by kLabz**
Also allow setting -warn to other levels (0-4 instead of 0-1)

[Changes without whitespace](https://github.com/HaxeFoundation/hxcs/pull/40/files?w=1)

### CODE REVIEW

File: build-tool/src/compiler/cs/CsCustomDefine.hx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package compiler.cs;
+
+@:enum abstract CsCustomDefine(String) to String {
+	/**
+		If set, output assembly as a library dll.
+	**/
+	var DLL = 'dll';
```
Comment (Gama11) [inline build-tool/src/compiler/cs/CsCustomDefine.hx]: I think the naming convention for enum values is usually `PascalCase`, not `UPPER_CASE`? At least in the standard library, not sure if hxcs has other enums?

Comment (kLabz) [reply build-tool/src/compiler/cs/CsCustomDefine.hx]: Yeah I'm more into PascalCase for enum values too, I'll update.

Comment (kLabz) [reply build-tool/src/compiler/cs/CsCustomDefine.hx]: Well actually in the case of `DLL` I'll keep it uppercase since it's an acronym.

Comment (Gama11) [reply build-tool/src/compiler/cs/CsCustomDefine.hx]: Not sure I agree, `Http` for instance is usually not uppercased either.

https://stackoverflow.com/a/15526526/2631715

**Pull request closed without merging**